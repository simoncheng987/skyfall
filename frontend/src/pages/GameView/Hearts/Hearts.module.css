.heart {
  width: 30px;
}

.heartContainer {
  display: flex;
  overflow: hidden;
}

.inactive {
  color: #434343;
}

.active {
  color: #e53131;
}
