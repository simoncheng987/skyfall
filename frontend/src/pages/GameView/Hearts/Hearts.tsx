import React from 'react';
import { HeartIcon } from '@heroicons/react/solid';
import styles from './Hearts.module.css';

interface HeartsProps {
  totalHearts: number;
  remainingHearts: number;
  className?: string;
}

export default function Hearts({ totalHearts, remainingHearts, className }: HeartsProps) {
  return (
    <div className={`${styles.heartContainer} ${className}`}>
      {Array.from({ length: totalHearts - remainingHearts }, () => (
        <HeartIcon className={`${styles.heart} ${styles.inactive}`} />
      ))}
      {Array.from({ length: remainingHearts }, () => (
        <HeartIcon className={`${styles.heart} ${styles.active}`} />
      ))}
    </div>
  );
}

Hearts.defaultProps = {
  className: '',
};
