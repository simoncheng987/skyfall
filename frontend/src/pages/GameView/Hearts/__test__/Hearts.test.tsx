import React from 'react';
import renderer from 'react-test-renderer';
import Hearts from '../Hearts';

describe('Hearts', () => {
  const className = 'testClassName';

  it('matches total hearts and remaining hearts', () => {
    const tree = renderer.create(<Hearts totalHearts={5} remainingHearts={3} />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with className prop', () => {
    const tree = renderer
      .create(<Hearts totalHearts={5} remainingHearts={3} className={className} />)
      .toTree();
    expect(tree).toMatchSnapshot();
  });
});
