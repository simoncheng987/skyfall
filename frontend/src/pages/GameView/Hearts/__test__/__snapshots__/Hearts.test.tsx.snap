// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hearts matches snapshot with className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "testClassName",
    "remainingHearts": 3,
    "totalHearts": 5,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <ForwardRef(HeartIcon)
            className="heart inactive"
          />,
          <ForwardRef(HeartIcon)
            className="heart inactive"
          />,
        ],
        Array [
          <ForwardRef(HeartIcon)
            className="heart active"
          />,
          <ForwardRef(HeartIcon)
            className="heart active"
          />,
          <ForwardRef(HeartIcon)
            className="heart active"
          />,
        ],
      ],
      "className": "heartContainer testClassName",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart inactive",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart inactive",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart active",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart active",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart active",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;

exports[`Hearts matches total hearts and remaining hearts 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "className": "",
    "remainingHearts": 3,
    "totalHearts": 5,
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <ForwardRef(HeartIcon)
            className="heart inactive"
          />,
          <ForwardRef(HeartIcon)
            className="heart inactive"
          />,
        ],
        Array [
          <ForwardRef(HeartIcon)
            className="heart active"
          />,
          <ForwardRef(HeartIcon)
            className="heart active"
          />,
          <ForwardRef(HeartIcon)
            className="heart active"
          />,
        ],
      ],
      "className": "heartContainer ",
    },
    "rendered": Array [
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart inactive",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart inactive",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart active",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart active",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "nodeType": "host",
        "props": Object {
          "aria-hidden": "true",
          "children": <path
            clipRule="evenodd"
            d="M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z"
            fillRule="evenodd"
          />,
          "className": "heart active",
          "fill": "currentColor",
          "viewBox": "0 0 20 20",
          "xmlns": "http://www.w3.org/2000/svg",
        },
        "rendered": Array [
          Object {
            "instance": null,
            "nodeType": "host",
            "props": Object {
              "clipRule": "evenodd",
              "d": "M3.172 5.172a4 4 0 015.656 0L10 6.343l1.172-1.171a4 4 0 115.656 5.656L10 17.657l-6.828-6.829a4 4 0 010-5.656z",
              "fillRule": "evenodd",
            },
            "rendered": Array [],
            "type": "path",
          },
        ],
        "type": "svg",
      },
    ],
    "type": "div",
  },
  "type": [Function],
}
`;
